src/meta/greedy_load_balancer.cpp

```diff
@@ -936,12 +936,37 @@ void greedy_load_balancer::cluster_balancer()
             return;
         }
         if (!t_migration_result->empty()) {
-            ddebug("migration count of move primary = {}", t_migration_result->size());
+            ddebug_f("migration count of move primary = {}", t_migration_result->size());
             return;
         }
     }
 
-    // TODO(zlw): copy secondary
+    bool need_continue = cluster_replica_balance(cluster_balance_type::Secondary);
+    if (!need_continue) {
+        return;
+    }
+
+    // TODO(zlw): copy primary
+}
+
+bool greedy_load_balancer::cluster_replica_balance(const cluster_balance_type type)
+{
+    bool enough_information = do_cluster_replica_balance(type);
+    if (!enough_information) {
+        return false;
+    }
+    if (!t_migration_result->empty()) {
```

Comment:
If the migration result is not empty, why it will return false? If this balance type is copy_secondary, and migration result is not empty, it won't copy_primary.

Reply:
The return value means if it need to do next step. There is no need copy primary if the migration result of copy secondary is not null.